### ISSUE TITLE
How to implement provide context on global error handlers

### ISSUE THREAD

**Issue by theduyetmai** [apollographql/apollo-server#1681]
I just check the below PR but still do not know how to implement. Any else can help me?
https://github.com/apollographql/apollo-server/issues/1343

Thank you so much

**Comment by Whoaa512** [comment]
@theduyetmai, if you use the [`extensions` API](https://www.apollographql.com/docs/apollo-server/features/metrics.html#Granular-Logs) (which is not well documented, so I suggest reading the source). You can create a custom error tracking extension that will be passed the `context` to each of the lifecycle methods.

```javascript
const { ApolloServer } = require('apollo-server')

import { GraphQLExtension } from 'graphql-extensions'

class MyErrorTrackingExtension extends GraphQLExtension {
    willSendResponse(o) {
        const { context, graphqlResponse } = o

        context.trackErrors(graphqlResponse.errors)

        return o
    }
    // Other lifecycle methods include
    // requestDidStart
    // parsingDidStart
    // validationDidStart
    // executionDidStart
    // willSendResponse
}

const server = new ApolloServer({
    typeDefs,
    resolvers,
    extensions: [() => new MyErrorTrackingExtension()],
    context({ req }) {
        return {
            req,
            trackErrors(errors) {
                // Track the errors
            },
        }
    },
})

server.listen().then(({ url }) => {
    console.log(`🚀  Server ready at ${url}`)
})
```

**Comment by theduyetmai** [comment]
Thank you so much

**Comment by kosanna** [comment]
on which version of apollo-server it's supposed to work? I have 2.1.0, but still have context being undefined

**Comment by Whoaa512** [comment]
@kosanna According to the release notes it should have gone out in [v2.0.8](https://github.com/apollographql/apollo-server/blob/master/CHANGELOG.md#v208)

**Comment by omgrief** [comment]
Bumping this issue, the fix brought in to v2.0.8 does seem like it would be the fix to allow for the context to be accessible from within the `formatError` function, but it is not the case as it never gets passed to the actual formatter, and is just returned by the extension.
